code/game/objects/effects/decals/Cleanable/tar.dm

```diff
@@ -0,0 +1,47 @@
+/obj/effect/decal/cleanable/tar
+	gender = PLURAL
+	name = "tar"
+	desc = "A sticky substance."
+	icon_state = "tar2"
+	density = FALSE
+	opacity = FALSE
+	var/turf/simulated/target
+	var/old_slowdown
+
+/obj/effect/decal/cleanable/tar/New(turf/simulated/loc)
+	if(!istype(loc, /turf/simulated) || loc.has_tar) // Check if the turf already has tar
+		qdel(src)
+		return
+	loc.has_tar = TRUE
+	target = loc
+	old_slowdown = target.slowdown // Store the original slowdown value
+	target.slowdown += 10 // Apply the slowdown effect to the turf
+	RegisterSignal(target, COMSIG_COMPONENT_CLEAN_ACT, PROC_REF(remove_tar))
+
+
+	if(prob(50))
+		icon_state = "tar3"
+	..()
+
+/obj/effect/decal/cleanable/tar/proc/remove_tar(datum/source)
+	SIGNAL_HANDLER
+	if(target) // Check if the target turf is valid
+		target.slowdown = old_slowdown
+		target.has_tar = FALSE
+		qdel(src)
+
+/obj/effect/decal/cleanable/tar/Crossed(AM as mob|obj)
```

Comment:
Can you typecast this instead to /atom/movable, and rename the variable name?
```suggestion
/obj/effect/decal/cleanable/tar/Crossed(AM)
```

Reply:
Typecast it, but Crossed is defined elsewhere and can't be renamed and AM seems really standardized and shouldn't need renaming.

Reply:
Renaming the variable won't do anything different/break anything. I still think it'd be easier to parse. As for the `as mob|obj` we just don't do that anymore.